fendermint/vm/interpreter/src/fvm/checkpoint.rs

```diff
@@ -87,11 +87,66 @@ where
     }
 }
 
+/// Sign the current and any incomplete checkpoints.
+pub async fn broadcast_incomplete_signatures<C, DB>(
+    client: &C,
+    validator_ctx: &ValidatorContext<C>,
+    gateway: &GatewayCaller<DB>,
+    chain_id: ChainID,
+    incomplete_checkpoints: Vec<getter::BottomUpCheckpoint>,
+) -> anyhow::Result<()>
+where
+    C: Client + Clone + Send + Sync + 'static,
+    DB: Blockstore + Send + Sync + 'static,
+{
+    for cp in incomplete_checkpoints {
+        let height = Height::try_from(cp.block_height)?;
+        let power_table = get_power_table(client, height)
+            .await
+            .context("failed to get power table")?;
+
+        if let Some(validator) = power_table
+            .0
+            .iter()
+            .find(|v| v.public_key.0 == validator_ctx.public_key)
+            .cloned()
+        {
+            // TODO: Code generation in the ipc-solidity-actors repo should cater for this.
```

Comment:
@cryptoAtwill, can you double-check that this is the case once you migrate to the `ipc-solidity-actors` bindings?

Reply:
I don't remember seeing that this has been implemented already, we just talked about the fact that the bindings generate as many versions of `SubnetID` and whatnot as there are facets. So it's just a note for the future that if this gets out of hand, we should try to solve it there.